### PR TITLE
[nimsuggest] fix def call on identifier 2 times on the line

### PR DESCRIPTION
**Pull request by yyoncho**
- apparently TLineInfo's implementation of `==` ignores the column. After I fixed
the code to use exact TLineInfo comparison I fixed several other issues hidden
by that issue.

- Replaced `tuple[sym, info]` with `SymInfoPair`

### CODE REVIEW

File: compiler/modulegraphs.nim [hunk 1]
```diff
@@ -53,6 +53,10 @@ type
     concreteTypes*: seq[FullId]
     inst*: PInstantiation
 
+  SymInfoPair* = ref object
```
Comment (Araq) [inline compiler/modulegraphs.nim]: Why is this a `ref object`? It should be just `object` as far as I can tell from how it's used.

Comment (yyoncho) [reply compiler/modulegraphs.nim]: Fixed.

**Comment by narimiran** [comment]
Should this be backported to Nim 1.6.x?

**Comment by yyoncho** [comment]
@narimiran yes. I would say everything nimsuggest related should be backported (unless specified otherwise)

**Pull request merged**